roles/openshift_sanitize_inventory/tasks/main.yml

```diff
@@ -77,6 +77,23 @@
         openshift_cloudprovider_vsphere_host is undefined or openshift_cloudprovider_vsphere_datacenter is undefined or
         openshift_cloudprovider_vsphere_datastore is undefined )
 
+- name: Ensure azure configuration variables are defined
+  block:
+    - name: ensure provider configuration variables are defined
+      fail:
+        msg: "'Required variable for azure provider '{{ item }}' is not defined"
+      when: vars[item] is not defined
+      with_items:
+        - openshift_cloudprovider_azure_client_id
+        - openshift_cloudprovider_azure_client_secret
+        - openshift_cloudprovider_azure_tenant_id
+        - openshift_cloudprovider_azure_subscription_id
+        - openshift_cloudprovider_azure_resource_group
+        - openshift_cloudprovider_azure_location
+  when:
```

Comment:
We like to put the when conditions at the start of the block, much easier to read that way.